actions/db_run_result_iterator.go

```diff
@@ -57,19 +57,19 @@ func (it *DBRunResultIterator) maybeFetch() {
 	if len(it.buf) > 0 {
 		return
 	}
-
+	if it.state == iteratorStateInit {
```

Comment:
Why do you need Init & Query states?
I understand where `iteratorStateDone` is being taking care of, but I don't see the difference between the 2 states above, now that lines 69-72 are deleted.

Reply:
right, it made sense when we supported seek as we like to have >= for the first call and > for the rest.
will remove the use of this state